fast/stages/02-networking-nva/README.md

```diff
@@ -0,0 +1,343 @@
+
+The VPCs are connected through two sets of sample NVA machines, grouped by region in [Managed Instance Groups (MIGs)](https://cloud.google.com/compute/docs/instance-groups). The appliances are plain Linux machines, performing simple routing/natting, leveraging some standard Linux features, such as *ip route* or *iptables*. While they are suited for demo purposes only, they should be substituted with enterprise appliances, before moving to production.
+The traffic destined to the VMs in each MIG is mediated through internal load balancers in each region, both in the trusted and in the untrusted networks.
+
+By default, the design assumes the following:
+
+- on-prem networks (and related resources) are considered trusted. As such, the VPNs with on-prem are terminated in GCP on the trusted VPC
+- the public Internet is considered untrusted. As such [Cloud NAT](https://cloud.google.com/nat/docs/overview) has been deployed in the untrusted landing VPC only
+- any traffic from an untrusted network to a trusted network (and vice-versa), and cross-environment traffic passes through the NVAs. For demo purposes the current NVA performs simple routing/natting only
+- any traffic from a trusted network to an untrusted network (e.g. Internet) is natted by the appliances. Futher exclusions can be anyway configured ad-hoc
+
+The trusted landing VPC acts as a hub and hosts the external connectivity and the shared services consumed by the spoke VPCs, which are connected to it via VPC network peerings. Spokes are used to partition environments, which is a fairly common pattern:
+
+- one spoke VPC for the development environment
+- one spoke VPC for the production environment
+
+Each VPC is created into its own project, and each project is configured as a Shared VPC host, so that network-related resources and access configurations via IAM are kept separate for each VPC.
+
+The design can be easily extended to host additional environments, or adopting a different logical mapping for spokes (e.g. one spoke for each company entity, etc.). Adding spokes is a trivial operation and it does not increase the design complexity. Operational processes to add more spokes are further explained in the following sections.
+
+In multi-organization scenarios, where production and non-production resources use different Cloud Identity and GCP organizations, the hub/landing VPC is usually part of the production organization. It establishes connections with the production spokes within its same organization, and with non-production spokes in a different organization.
+
+An additional VPC disconnected from the rest of the networks is deployed with this code. It hosts two VMs emulating the on-prem environment for testing purposes, via a Docker network and containers (for VPN, DNS, HTTP, etc).
```

Comment:
Is it?

Reply:
I just kept it form the 02-networking stage. I thought it was also coming here. Never mind :) I'll remove it!